src/commands/youtube.ts

```diff
@@ -0,0 +1,215 @@
+import { DataService } from '../tasks/Promotions/YouTube/DataService';
+import { rolesIds } from '../roles';
+
+export class YouTubeCommand extends SlashCommand {
+  private allowedRoles: string[]
+
+  constructor(creator: SlashCreator) {
+    super(creator, {
+      name: 'youtube',
+      description: 'YouTube realated commands',
+      guildIDs: [process.env.GUILD_ID],
+      options: [
+        {
+          name: 'subscribe',
+          description: 'Subscribe to a YouTube channel',
+          type: CommandOptionType.SUB_COMMAND,
+          options: [
+            {
+              type: CommandOptionType.STRING,
+              name: 'name',
+              description: 'channel name',
+              required: true
+            },
+            {
+              type: CommandOptionType.STRING,
+              name: 'channel_id',
+              description: 'channel ID',
+              required: true
+            }
+          ]
+        },
+        {
+          name: 'unsubscribe',
+          description: 'Unsubscribe from a YouTube channel',
+          type: CommandOptionType.SUB_COMMAND,
+          options: [
+            {
+              type: CommandOptionType.STRING,
+              name: 'name',
+              description: 'channel name',
+              required: true
+            },
+          ]
+        },
+        {
+          name: 'find',
+          description: 'Check channel subscription',
+          type: CommandOptionType.SUB_COMMAND,
+          options: [
+            {
+              type: CommandOptionType.STRING,
+              name: 'name',
+              description: 'channel name',
+              required: true
+            },
+          ]
+        }
+      ]
+    });
+
+    // Not required initially, but required for reloading with a fresh file.
+    this.filePath = __filename;
+    this.allowedRoles = [
+      rolesIds.admins,
+      rolesIds.leadAdmin,
+      rolesIds.leadAmbassador,
+      rolesIds.leadDev
+    ]
+  }
+
+  async run(ctx: CommandContext): Promise<void> {
+    try {
+
+      if (!this.hasAllowedRoles(ctx)) {
+        throw {
+          title: 'Access denied',
+          message: `You do not have access to this command.`
+        }
+      }
+
+      const { subscribe, unsubscribe, find } = ctx.options;
+
+      if (subscribe) {
+        await this.subscribe(ctx);
+      }
+
+      if (unsubscribe) {
+        await this.unsubscribe(ctx);
+      }
+
+      if (find) {
+        await this.find(ctx);
+      }
+
+    } catch(e) {
+      console.error(e);
+      const embed = this.createStatusEmbed(
+        e.title || 'Error',
+        e.message || 'Something went wrong - try again later...',
+        true
+      );
+
+      await ctx.send({embeds: [embed], ephemeral: true});
+    }
+  }
+
+  async subscribe(ctx: CommandContext): Promise<void> {
+    const { name, channel_id } = ctx.options.subscribe;
+
+    const recordByName = await DataService.getChannelByName(name);
+
+    if (recordByName) {
```

Comment:
I learned from Justin that it's not best practice to rely on truthy tests.  It's better to do something like this:

```javascript
 if (recordByName !== undefined) ...
```

Reply:
Here I get an issue.
<img width="460" alt="Screenshot 2021-08-31 at 21 44 13" src="https://user-images.githubusercontent.com/41327204/131566062-221ca8f4-cec8-4e36-8860-f7bcbf4e7911.png">

Reply:
> I learned from Justin that it's not best practice to rely on truthy tests. It's better to do something like this:
> 
> ```js
>  if (recordByName !== undefined) ...
> ```

in this case, I guess better to make:
```js
if (recordByName != undefined)
```
then it will cover and undefined and null

Reply:
my noob mistake 🤣